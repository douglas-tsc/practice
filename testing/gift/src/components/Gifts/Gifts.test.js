import React from 'react'
import Gift from './Gifts'
import { shallow } from 'enzyme'

describe('Gifts', () => {
  const mockRemove = jest.fn()

  const props = { id: 1, removeGift: mockRemove }
  const gift = shallow(<Gift {...props} />)

  it('renders correctly', () => {
    expect(gift).toMatchSnapshot()
  })

  it('initializes a person and present in state', () => {
    expect(gift.state()).toEqual({ person: '', present: '' })
  })

  describe('When typing into the person input', () => {
    const person = 'Uncle'
    beforeEach(() => {
      gift
        .find('[data-test="addPersonName"]')
        .simulate('change', { target: { value: person } })
    })

    it('updates the person in state', () => {
      expect(gift.state().person).toEqual(person)
    })
  })

  describe('When typing into the present input', () => {
    const present = 'Chocolates'
    beforeEach(() => {
      gift
        .find('[data-test="addPresentName"]')
        .simulate('change', { target: { value: present } })
    })

    it('updates the person in state', () => {
      expect(gift.state().present).toEqual(present)
    })
  })

  describe('When clicking the remove button...', () => {
    beforeEach(() => {
      gift.find('[data-test="removeButton"]').simulate('click')
    })
    it('...calls the remove gifts callback', () => {
      expect(mockRemove).toHaveBeenCalledWith(1)
    })
  })
})
