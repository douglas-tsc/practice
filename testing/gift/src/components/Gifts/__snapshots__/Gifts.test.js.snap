// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gifts renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Gift
    id={1}
    removeGift={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          data-test="addPersonName"
          onChange={[Function]}
          placeholder="Person"
          type="text"
        />,
        ";",
        <input
          data-test="addPresentName"
          onChange={[Function]}
          placeholder="Present"
          type="text"
        />,
        <button
          data-test="removeButton"
          onClick={[Function]}
        >
          Remove
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "data-test": "addPersonName",
          "onChange": [Function],
          "placeholder": "Person",
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      ";",
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "data-test": "addPresentName",
          "onChange": [Function],
          "placeholder": "Present",
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Remove",
          "data-test": "removeButton",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Remove",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            data-test="addPersonName"
            onChange={[Function]}
            placeholder="Person"
            type="text"
          />,
          ";",
          <input
            data-test="addPresentName"
            onChange={[Function]}
            placeholder="Present"
            type="text"
          />,
          <button
            data-test="removeButton"
            onClick={[Function]}
          >
            Remove
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "data-test": "addPersonName",
            "onChange": [Function],
            "placeholder": "Person",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        ";",
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "data-test": "addPresentName",
            "onChange": [Function],
            "placeholder": "Present",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Remove",
            "data-test": "removeButton",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Remove",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
